### PR TITLE
fixed 2 memory leak issues

### PR DESCRIPTION
**Pull request by crackevil**
fixed 2 memory leak issues which are not often occur.
please have them reviewed and commit back to the HG server
Thanks

### CODE REVIEW

File: chrome/content/ui/sidebar.js [hunk 1]
```diff
@@ -455,16 +455,9 @@ function fillInContext(/**Event*/ e)
...
 function handleClick(event)
 {
-  let item = treeView.getItemAt(event.clientX, event.clientY);
-  if (event.button == 0 && treeView.getColumnAt(event.clientX, event.clientY) == "state")
-  {
-    let filter = getFilter(item);
-    if (filter)
-      enableFilter(filter, filter.disabled);
-    event.preventDefault();
-  }
```
Comment (palant) [inline chrome/content/ui/sidebar.js:465]: I don't see any memory leak in this code (nor can I reproduce one triggered here), you removed useful functionality however. Could you explain how you reproduced the leak?

File: chrome/content/ui/filters-subscriptionview.js [hunk 2]
```diff
@@ -288,18 +287,24 @@ ListManager.prototype =
...
                   SubscriptionActions.updateCommands);
   E("acceptableAds").checked = FilterStorage.subscriptions.some(s => s.url == Prefs.subscriptions_exceptionsurl);
   E("acceptableAds").parentNode.hidden = !ListManager.acceptableAdsCheckbox;
 };
 
+ListManager.uninit = function()
+{
+	ListManager.lm1=null;
+	ListManager.lm2=null;
+}
```
Comment (palant) [inline chrome/content/ui/filters-subscriptionview.js:307]: Same issue as well - this shouldn't be necessary, Firefox will take care of garbage collecting all objects once the window is closed. How did you reproduce a memory leak here?

**Comment by palant** [comment]
Sorry about the late response - I wanted to look into this later but forgot to mark as unread.

**Comment by crackevil** [comment]
for both leaks, you can reproduce them as follow,(use side.xul one as example)
prepare:
install an addon named "about:ccdump", linked https://addons.mozilla.org/en-US/firefox/addon/cycle-collector-analyzer/

how to reproduce:
1. open "about:ccdump", click analysis, search "sidebar.xul", you will found nothing, of course. close the tab "about:ccdump"
2. open a web page
3. open AdblockPlus's side bar aka, blockable items
4. click any item in the side bar
5. close side bar
6. open "about:ccdump" again, click analysis, search "sidebar.xul", you will found a lot of orphan xul nodes of sidebar.xul

expected:
do step 1~5, you should see nothing of "sidebar.xul" kept in memory.

# 

I guess the event of click conflicts the handling process of onselect caused this memory leak.

**Comment by palant** [comment]
Just because the page is closed, it doesn't mean that it has to be garbage collected immediately. You should open `about:memory` and click "Minimize memory usage". And even that doesn't always work lately for some reason, sometimes you just have to wait...

**Pull request closed without merging**